config/libc/newlib.in.2

```diff
@@ -65,6 +65,77 @@ config LIBC_NEWLIB_NANO_FORMATTED_IO
       with size constraint issues.  This option does not affect wide-char
       formatted I/O functions.
 
+config LIBC_NEWLIB_REGISTER_FINI
+    bool
+    prompt "Enable Register finalization"
+    depends on LIBC_NEWLIB_2_2_or_later
+    help
+      Enable finalization function registration using atexit.
+
+config LIBC_NEWLIB_REENT_SMALL
+    bool
+    prompt "Enable Small reent structure"
+    depends on LIBC_NEWLIB_2_2_or_later
+    help
+      Enable small reentrant struct support.
+
+config LIBC_NEWLIB_ATEXIT_DYNAMIC_ALLOC
+    bool
+    prompt "Disable atexit dynamic allocation"
```

Comment:
The name of the option and the prompt are counter-intuitively named. One would expect the check for `"${CT_LIBC_NEWLIB_ATEXIT_DYNAMIC_ALLOC} = y` to verify that the "atexit dynamic allocation is enabled".

There are similar misnomers below.

I think the best approach to keep the option names, add `default y`, and correct prompts/help messages; and invert the checks in newlib.sh.

Alternatively, the options may be renamed to LIBC_NEWLIB_NO_ATEXIT_DYNAMIC_ALLOC, etc.